src/functions-and-arrays.js

```diff
@@ -1,41 +1,128 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(a, b) {
+  if ( a >= b ) {
+    return a;
+  } else {
+    return b;
+  }
+}
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(array) {
+  let longest;
+  if ( array.length === 0 ) { 
+    return null
+  } else {
+    for ( let i = 0; i < array.length; i++ ) {
+      const word = array[i];
+      if ( !longest || longest.length < word.length ) {
+        longest = word;
+      }
+    }
+    return longest;
+  }
+}
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
-
+function sumNumbers(array) {
+  let total = 0;
+  if ( array.length === 0 ) {
+    return 0;
+  } else {
+    for ( let i = 0; i < array.length; i++ ) {
+      total += array[i]
+    } 
+  }
```

Comment:
```suggestion
    for ( let i = 0; i < array.length; i++ ) {
      total += array[i]
    }
```
Esta perfect lo que has peusto pero piensa lo siguiente, cuando va a entrar en el bucle hace la siguiente comprobacion let i = 0; i < 0; i++ (en el caso de que el length sea 0) es decir que no entraria en el bucle!, pero lo que has puesto tu esta perfect te aseguras vamos pero el resultado seria el mismo y menos lineas de codigo! :D